packages/next/src/shared/lib/get-img-props.ts

```diff
@@ -67,7 +67,10 @@ export type ImageProps = Omit<
    * @deprecated This prop does not do anything.
    */
   lazyRoot?: string
-}
+} & (
```

Comment:
Very clever pattern! 👏 The only suggestion I have is to add a brief comment for clarity, as this logic might be a bit hard to follow for some developers. For example:

```ts
/**
 * This logic ensures:
 * - When `fill` is `true`, `width` and `height` are optional and can be either numbers or numeric strings.
 * - When `fill` is `false` (or omitted), `width` and `height` are required and follow the same type rules.
 */
```

Reply:
Thanks 🙌 Of course I added and committed

Reply:
Well done :)